src/alexapi/playback_handlers/soxhandler.py

```diff
@@ -0,0 +1,107 @@
+import subprocess
+from basehandler import BaseHandler, PlaybackAudioType
+
+MAX_VOLUME_GAIN = 30
+
+
+class SoxHandler(BaseHandler):
+	def __init__(self, config, callback_report):
+		super(SoxHandler, self).__init__(config, callback_report)
+		self.__config = config
+
+		self.volume_gain = 0
+		self.media_volume_gain = 0
+
+		self.parameters_speech = []
+		self.parameters_media = []
+
+		self.playback_padding = 0
+
+		self.proc = None
+
+	def on_setup(self):
+		if self.__config['sound']['output']:
+			self.parameters_speech.extend(['-t', self.__config['sound']['output']])
+			if self.__config['sound']['output_device']:
+				self.parameters_speech.append(self.__config['sound']['output_device'])
+
+		if self.__config['sound']['media_output']:
+			self.parameters_media.extend(['-t', self.__config['sound']['media_output']])
+			if self.__config['sound']['media_output_device']:
+				self.parameters_media.append(self.__config['sound']['media_output_device'])
+		else:
+			self.parameters_media = self.parameters_speech
+
+		if self.__config['sound']['default_volume']:
+			self.on_set_volume(self.__config['sound']['default_volume'])
+		if self.__config['sound']['media_default_volume']:
+			self.on_set_media_volume(self.__config['sound']['media_default_volume'])
+
+		self.playback_padding = str(self.__config['sound']['playback_padding'])
+
+	def on_play(self, item):
+		# SoX can't play file URLs
+		audio_file = item.url.replace('file://', '')
+
+		sox_cmd = ['sox', '-q', audio_file]
+		if item.audio_type == PlaybackAudioType.SPEECH:
+			sox_cmd.extend(self.parameters_speech)
+			sox_cmd.extend(['vol', str(self.volume_gain), 'dB'])
+		else:
+			sox_cmd.extend(self.parameters_media)
+			sox_cmd.extend(['vol', str(self.media_volume_gain), 'dB'])
+
+		if item.offset:
+			sox_cmd.extend(['trim', self.__calculate_offset(item.offset)])
+
+		sox_cmd.extend(['pad', self.playback_padding, self.playback_padding])
```

Comment:
Some SoX effects like padding can be applied both at the beginning and end of the audio. In this case it seemed in testing that padding the beginning and end of audio playback offered a cleaner transition than not. Given this padding effect is generally a hack for platforms not performing as well as others yet it seemed reasonable